apps/server/src/modules/room/api/room-invitation-link.controller.ts

```diff
@@ -0,0 +1,111 @@
+	): Promise<RoomInvitationLinkResponse> {
+		const roomInvitationLink = await this.roomInvitationLinkUc.createLink(currentUser.userId, createRoomParams);
+
+		const response = RoomInvitationLinkMapper.mapToRoomInvitationLinkResponse(roomInvitationLink);
+
+		return response;
+	}
+
+	@Put(':roomInvitationLinkId')
+	@ApiOperation({ summary: 'Update an existing room invitation link' })
+	@ApiResponse({
+		status: HttpStatus.OK,
+		description: 'Returns the updated room invitation link',
+		type: RoomInvitationLinkResponse,
+	})
+	@ApiResponse({ status: HttpStatus.BAD_REQUEST, type: ApiValidationError })
+	@ApiResponse({ status: HttpStatus.UNAUTHORIZED, type: UnauthorizedException })
+	@ApiResponse({ status: HttpStatus.FORBIDDEN, type: ForbiddenException })
+	@ApiResponse({ status: HttpStatus.NOT_FOUND, type: NotFoundException })
+	@ApiResponse({ status: '5XX', type: ErrorResponse })
+	public async updateLink(
+		@CurrentUser() currentUser: ICurrentUser,
+		@Param() urlParams: RoomInvitationLinkUrlParams,
+		@Body() updateRoomParams: UpdateRoomInvitationLinkBodyParams
+	): Promise<RoomInvitationLinkResponse> {
+		const roomInvitationLink = await this.roomInvitationLinkUc.updateLink(currentUser.userId, {
+			id: urlParams.roomInvitationLinkId,
+			...updateRoomParams,
+		});
+
+		const response = RoomInvitationLinkMapper.mapToRoomInvitationLinkResponse(roomInvitationLink);
+
+		return response;
+	}
+
+	@Delete(':roomInvitationLinkId')
+	@ApiOperation({ summary: 'Delete a room invitation link' })
+	@ApiResponse({ status: HttpStatus.NO_CONTENT, description: 'Deletion successful' })
+	@ApiResponse({ status: HttpStatus.BAD_REQUEST, type: ApiValidationError })
+	@ApiResponse({ status: HttpStatus.UNAUTHORIZED, type: UnauthorizedException })
+	@ApiResponse({ status: HttpStatus.FORBIDDEN, type: ForbiddenException })
+	@ApiResponse({ status: HttpStatus.NOT_FOUND, type: NotFoundException })
+	@ApiResponse({ status: '5XX', type: ErrorResponse })
+	@HttpCode(204)
+	public async deleteLink(
+		@CurrentUser() currentUser: ICurrentUser,
+		@Param() urlParams: RoomInvitationLinkUrlParams
+	): Promise<void> {
+		await this.roomInvitationLinkUc.deleteLink(currentUser.userId, urlParams.roomInvitationLinkId);
+	}
+
+	@Post(':roomInvitationLinkId')
+	@ApiOperation({ summary: 'Use a room invitation link to join a room' })
+	@ApiResponse({ status: HttpStatus.OK })
+	@ApiResponse({ status: HttpStatus.BAD_REQUEST, type: ApiValidationError })
+	@ApiResponse({ status: HttpStatus.UNAUTHORIZED, type: UnauthorizedException })
+	@ApiResponse({ status: HttpStatus.FORBIDDEN, type: ForbiddenException })
+	@ApiResponse({ status: '5XX', type: ErrorResponse })
+	public async useLink(
+		@CurrentUser() currentUser: ICurrentUser,
+		@Param() urlParams: RoomInvitationLinkUrlParams
+	): Promise<string> {
+		const roomId = await this.roomInvitationLinkUc.useLink(currentUser.userId, urlParams.roomInvitationLinkId);
+
+		return roomId;
```

Comment:
why do we want to return the roomId of the link here now?

Reply:
because we want the frontend to be able to redirect to the room. Just realized that this is missing during the store implementation.